py/visdom/__init__.py

```diff
@@ -2209,3 +2209,75 @@ def mesh(self, X, Y=None, win=None, env=None, opts=None):
             'layout': _opts2layout(opts),
             'opts': opts,
         })
+    @pytorch_wrap
+    def double_yaxis_lines(self,X=None,Y1=None,Y2=None,opts=None,win=None,env=None):
+        '''
+        This function will create a line plot using plotly with different Y-Axis.
+        `X`  = A numpy array of the range.
+        `Y1` = A numpy array of the same count as X.
+        `Y2` = A numpy array of the same count as X.
+        Here are the `opts` options:
+        - `opts.height` : Height of the plot
+        - `opts.width` :  Width of the plot
+        - `opts.nameY1` : Axis name for Y1 plot
+        - `opts.nameY2` : Axis name for Y2 plot
+        - `opts.title` :  Title of the plot
+        - `opts.color_title_Y2` :  Color of the axis Title
+        - `opts.color_tick_Y2` :  Color of the axis Ticks
+        - `opts.side` :  side on which the tick has to be placed. Has values 'right' or `left`.
+        - `opts.showlegend` :  Display legends (boolean values)
+        - `opts.top` :  Set the top margin of the plot
+        - `opts.bottom` :  Set the bottom margin of the plot
+        - `opts.right` :  Set the right margin of the plot
+        - `opts.left` :  Set the left margin of the plot   
+        '''
+        X = np.asarray(X)
+        Y1 = np.asarray(Y1)
+        Y2 = np.asarray(Y2)
+        assert X is not None,'X Cannot be None'
+        assert Y1 is not None,'Y1 Cannot be None'
+        assert Y2 is not None,'Y2 Cannot be None'
+        assert X.shape == Y1.shape,'values of X and Y1 are not in proper shape'
+        assert X.shape == Y2.shape,'values of X and Y2 are not in proper shape'
+        if opts == None:
+            opts={}
+            opts['height'] = 300
+            opts['width'] = 500
+        X = [float(value) for value in X]
+        Y1 = [float(value) for value in Y1]
+        Y2 = [float(value) for value in Y2]
+        trace1 = dict(
```

Comment:
nit: style. For all dicts, prefer the explicit construction, rather than:
```
dict(
          x= X,
          y= Y1,
          name= opts.get('nameY1','Y1 axis'),
          type= 'scatter'
)
```
use
```
{
    'x': X,
    'y': Y1,
    'name': opts.get('name_y1', 'Y1 axis'),
    'type': 'scatter',
}
```

Reply:
Changes Done!

Reply:
Almost there! For this dict and other layered dicts as well, the spacing should be 4 indents for each additional layer, not necessarily based on the indentation of the `{`. For example:
```
my_dict = {
    'value_1': 'a value',
    'value_2': 33,
    'embedded_dict': {
        'internal_val': 'something else!',
        'another_dict': {
            'internal_key': '3 layers of indent!',
        },
    },
    'another_key': 'another_value',
}
```

This style of formatting is generally the standard `PEP` style, which you might want to familiarize yourself more as you start contributing to python projects.

Reply:
Changes Done!.... 
I used this: https://codebeautify.org/python-formatter-beautifier
Some else if needed will be benefited.... I am bit lazy in case of formatting...... 😛
😀